src/bp-search/classes/class-bp-search.php

```diff
@@ -253,7 +253,9 @@ public function ajax_search() {
 			}
 
 			$args = array(
-				'search_term'   => $_REQUEST['search_term'],
+
+				// Needed to replace '&' with '&amp;' so it matches the data in db. '&' symbol usually be replaced with '&amp;' when saving in database.
+				'search_term'   => str_replace('&', '&amp;', $_REQUEST['search_term']),
```

Comment:
@kent-carlo Instead of `str_replace` you should use `htmlspecialchars` function to encode HTML. 
I'm assuming we need to replace all HTML special chars except quote. 

```echo htmlspecialchars($str, ENT_NOQUOTES);```

You need to do some test with adding aal HTML special chars.

Reply:
@dipskakadiya I will try, thanks